src/client/src/tscSQLParser.c

```diff
@@ -5034,6 +5040,21 @@ int32_t handleNeOptr(tSqlExpr** rexpr, tSqlExpr* expr) {
   return TSDB_CODE_SUCCESS;
 }
 
+static void convertWhereStringCharset(tSqlExpr* pRight){
+  if(pRight->value.nType != TSDB_DATA_TYPE_BINARY){
+    return;
+  }
+
+  char newData[TSDB_MAX_NCHAR_LEN] = {0};
+  int len = 0;
+  if(!taosMbsToUcs4(pRight->value.pz, pRight->value.nLen, newData, TSDB_MAX_NCHAR_LEN, &len)){
+    tscError("nchar in where condition mbsToUcs4 error");
+  }
+  pRight->value.pz = realloc(pRight->value.pz, len);
```

Comment:
better check return value of realloc. If allocate failure, the original pointer will be overwritten by NULL thus cause memory leak.